docs/advanced/how-it-works.md

```diff
@@ -0,0 +1,136 @@
+
+> The `merkleRoot` will always be the same for all the documents wrapped together (at the same time). It will be different for documents wrapped separately.
+
+Now that our batch of documents have a common identifier and that we can prove (thanks to the merkle tree algorithm) that the `targetHash` of a document was used to create a specific `merkleRoot`, we can use the `merkleRoot` in our document store and issue it.
+
+#### Data Obfuscation
+
+Thanks to the way we compute `targetHash`, OpenAttestation allows for one to obfuscate data they don't want to make public. For this just need to compute the hash of a specific field and add it into the documents. Let's try it with the [CLI](/docs/component/open-attestation-cli) and the document above:
+
+```bash
+open-attestation filter ./path/to/file.json ./output.json name
+```
+
+The content of `output.json` will be:
+
+```json
+{
+  "version": "https://schema.openattestation.com/2.0/schema.json",
+  "data": {
+    "issuers": [
+      {
+        "identityProof": {
+          "type": "40caddff-5cd4-477d-adf4-48dcd0a2d761:string:DNS-TXT",
+          "location": "c15358f4-f0dc-41c8-abfb-0d030aae3233:string:imaginative-amber-ferret.sandbox.openattestation.com"
+        },
+        "name": "0de92429-f8d3-47a0-868f-154227a66f40:string:DEMO STORE",
+        "tokenRegistry": "89c1f33c-121d-4622-a561-12fb400f2f3f:string:0x8194648f40ED07F841fA357Bf52CBE8D6d7ce48D"
+      }
+    ]
+  },
+  "signature": {
+    "type": "SHA3MerkleProof",
+    "targetHash": "11d456db211d68cc8a6eac5e293422dec669b54812e4975497d7099467335987",
+    "proof": [],
+    "merkleRoot": "11d456db211d68cc8a6eac5e293422dec669b54812e4975497d7099467335987"
+  },
+  "privacy": {
+    "obfuscatedData": [
+      "9d22655fcee6bf3eb10ba280cfa40e662f004a819be0b64e2fe9d0cebba6788f"
+    ]
+  }
+}
+```
+
+The `name` field is not available anymore in the `data` object, and the hash associated to it is added into `privacy.obfuscatedData`.
+
+> More importantly, the document remains valid.
+
+The hash added is the one used when computing the [`targetHash`](#targethash). Like said above, we follow the same steps to compute a `targetHash` and verify that a document remained untouched, with one subtle difference: all the hashes available in `privacy.obfuscatedData` are added to the list of computed hashes from the data object before sorting them. That way we can rebuild the `targetHash` with obfuscated values.
+
+Thanks to data obfuscation a user can decide to selectively decide which data he wants to share.
+
+### Document Store
+
+As discussed above the `targetHash` and the `merkleRoot` of a document can be used for issuance. However, the `merkleRoot` is by far the most efficient way.
```

Comment:
Don't think we want to give people an idea that they can issue with targetHash, they cant when it's batched. Prefer to speak about 

The issuer can choose to issue documents individually or in a batch...

Reply:
done